ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/RateLimitingConfig.scala

```diff
@@ -6,11 +6,13 @@ package com.daml.platform.apiserver.configuration
 final case class RateLimitingConfig(
     maxApiServicesQueueSize: Int,
     maxApiServicesIndexDbQueueSize: Int,
+    maxHeapSpacePercentage: Int,
 )
 
 case object RateLimitingConfig {
   val default: RateLimitingConfig = RateLimitingConfig(
     maxApiServicesQueueSize = 10000,
     maxApiServicesIndexDbQueueSize = 1000,
+    maxHeapSpacePercentage = 85,
```

Comment:
As discussed with @mziolekda there should also be a parameter that maxThreasholdSize.  The actual threshold used would be max(maxHeapSpacePercentage * maxHeapSize, maxHeapSize - maxThresholdSize)